model/serviceFilter.php

```diff
@@ -0,0 +1,16 @@
+<?php
+require_once "filterDecorator.php";
+
+class ServiceFilter extends FilterDecorator
+{
+    public $sname;
+    public function __construct($Filter)
+    {
+        $this->Filter = $Filter;

```

Comment:
You can see an inconsistency in the name of `$filter` between here and line 14. Make `$this->Filter` and `$Filter` lowercase (`$filter`).